### PR TITLE
added scholarbatchimporter to override getIdentifier and check for confi...

### PR DESCRIPTION
**Pull request by ppound**
...gured namespaces

### CODE REVIEW

**Comment by ppound** [comment]
forgot to put the jira ticket the comments https://jira.duraspace.org/browse/ISLANDORA-914

File: includes/importer.inc [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    if (!module_exists('islandora_basic_collection')) {
+      return $default_namespace;
+    }
+    $parent = islandora_object_load($this->parentPid);
+    if (isset($parent['COLLECTION_POLICY'])) {
+      module_load_include('inc', 'islandora_basic_collection', 'collection_policy');
+      $policy_stream = $parent['COLLECTION_POLICY'];
+      $policy = new CollectionPolicy($policy_stream->content);
+
+      foreach ($policy->getContentModels() as $pid => $info) {
```
Comment (adam-vessey) [inline includes/importer.inc]: No need to iterate here... We know we're only concerned with `ir:citationCModel`.

File: includes/importer.inc [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+   * @return string
+   *   the namespace to use
+   */
+  protected function getNamespaceFromCollectionPolicy($default_namespace) {
+    if (!module_exists('islandora_basic_collection')) {
+      return $default_namespace;
+    }
+    $parent = islandora_object_load($this->parentPid);
+    if (isset($parent['COLLECTION_POLICY'])) {
+      module_load_include('inc', 'islandora_basic_collection', 'collection_policy');
```
Comment (adam-vessey) [inline includes/importer.inc]: No need to manually load this include... The `CollectionPolicy` class is already available via autoloading: https://github.com/Islandora/islandora_solution_pack_collection/blob/7.x/islandora_basic_collection.info#L10

File: includes/importer.inc [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+   *   a tuque object
+   * @param string $namespace
+   *   a default namespace.  We use this namespace if we can't find a Collection
+   * Policy or the collection solution pack is not installed.
+   *
+   * @return string
+   *   An identifier for the repo.
+   */
+  protected function getIdentifier(IslandoraTuque $tuque, $namespace) {
+    static $cm_namespace;
```
Comment (adam-vessey) [inline includes/importer.inc:17]: Should probably use `drupal_static()`.

Comment (adam-vessey) [inline includes/importer.inc:17]: Eh... I could live without it, though.

File: includes/importer.inc [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+      foreach ($policy->getContentModels() as $pid => $info) {
+        if ($pid == 'ir:citationCModel' && isset($info['namespace'])) {
+          return $info['namespace'];
+        }
+      }
+    }
+    return $default_namespace;
+  }
+}
```
Comment (adam-vessey) [inline includes/importer.inc]: Missing a newline.

File: includes/importer.inc [hunk 5]
```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * @file
+ * Defines ScholarBatchImporter.
+ */
+
+abstract class ScholarBatchImporter extends IslandoraBatchImporter {
+
+  /**
+   * Get an identifier.
```
Comment (adam-vessey) [inline includes/importer.inc]: This comment should be minimal (as in, the entire doxygen comment), as it's overriding an already-existing method...  Should probably just update the ancestor's comment, reference it here, and indicate what's happening differently.

**Pull request merged**